tests/dag/test_dag.py

```diff
@@ -212,29 +211,12 @@ def test_plot_validates_html_extension_if_d3(dag, tmp_directory):
 
 @pytest.mark.parametrize('backend', [None, 'd3'])
 def test_plot_with_d3_embed(dag, tmp_directory, monkeypatch, backend):
-    # simulate pygraphviz isnt installed
+    # simulate pygraphviz isn't installed
     monkeypatch.setattr(dag_plot_module, 'find_spec', lambda _: None)
     output = dag.plot(backend=backend)
 
-    # test the svg tag has the rendered content
-    assert '<svg id="dag" viewBox=' in output.data
-    # and the js message is hidden
-    assert '<div id="js-message" style="display: none;">' in output.data
-
-
-@pytest.mark.skipif(IS_WINDOWS_PYTHON_3_10, reason="requires < 3.10")
-def test_plot_with_d3_embed_error_if_missing_dependency(
-        dag, tmp_directory, monkeypatch):
-    monkeypatch.setattr(ploomber_util.importlib.util, 'find_spec',
-                        lambda _: None)
-
-    with pytest.raises(ImportError) as excinfo:
-        dag.plot(backend='d3')
-
-    expected = ("'requests-html' 'nest_asyncio' are "
-                "required to use 'embedded HTML with D3 backend'. "
-                "Install with: pip install 'requests-html' 'nest_asyncio'")
-    assert expected == str(excinfo.value)
+    # test path exists
+    assert len(output.src) != 0
```

Comment:
here we should test `assert Path(output.src).is_file()`. I ran the code, and it's breaking because we're removing the file when we bass the `"embed"` option, which is the default.

we have to modify this: https://github.com/ploomber/ploomber/blob/9c296ecfde0c8359631f8b09e6a8f4507e95863f/src/ploomber/dag/dag.py#L933 so it doesn't delete the file

Reply:
Disabling the deletion doesn't solve the issue entirely since the temp file is being saved in `{temp-folder}/tmp` folder and when I use Jupyter notebook server it fails to access and embed the file with the following error: `Not allowed to load local resource: <URL>` 

Saving the file in the working dir solves this issue. 

Is this solution sounds OK? 
The main downside here is that every time we'll run the `plot` command (when the format of the file is `html`) it will generate a file.

Reply:
where did you test it? I think Python creates a temp file in a directory where it can read and write, but I might be wrong.

Reply:
I've tested it on windows by running `jupyter notebook`. It creates a tmp file in default directory i.e : 'C:\Users\User\AppData\Local\Temp\tmp1p6cdvl7.html'.
The file exists but jupyer can't access it. If I move the file to the folder where I'm running the notebook it works.

![image](https://user-images.githubusercontent.com/103026689/188573342-a0ae8d3c-1b9f-4458-af26-c701b7e6de0f.png)

![image](https://user-images.githubusercontent.com/103026689/188573411-83dad83c-dd2f-4c49-a47c-b6c50411bd68.png)

Reply:
I see; ok, let's create it locally then.

Reply:
in the current working directory